agilerl/utils/minari_utils.py

```diff
@@ -0,0 +1,102 @@
+import os
+import minari
+from minari.storage.hosting import download_dataset
+from minari.storage.local import load_dataset
+from minari.storage.datasets_root_dir import get_dataset_path
+from collections import deque
+import h5py
+from agilerl.components.replay_buffer import ReplayBuffer
+import torch
+
+
+def load_minari_dataset(dataset_id):
+    
+    if dataset_id not in list(minari.list_remote_datasets().keys()):
+        raise KeyError("Enter a Valid Minari Dataset ID. check https://minari.farama.org/ for more details.")
+        
+    file_path = get_dataset_path(dataset_id)
+    
+    if os.path.exists(file_path):
+        print("download dataset: ", dataset_id)
+        download_dataset(dataset_id)
+    
+    data_path = os.path.join(file_path, "data", "main_data.hdf5")
+    minari_dataset = h5py.File(data_path, 'r')
```

Comment:
I think it should be possible to load the minari dataset without importing h5py, and this is preferable if possible, because we want to abstract the file format of minari away from the user.

Reply:
And if it's not possible, we need to add features to Minari such that it becomes possible.

Reply:
Is it possible to do this? I don't see importing h5py as a problem because if someone wants to use a custom dataset, then they can load it this way - unless there's a Minari wrapper for this?

Reply:
The rationale is that we may change Minari's backend dataset file format, and don't want dependent libraries' integrations to break if we do. I think `minari.load_dataset(dataset_id)` should work here.

Reply:
Can you confirm @shreyansjainn?

Reply:
@nicku-a @balisujohn yes, i think the `minari.load_dataset(dataset_id)` should work here, will make the changes and push the updates today.

Reply:
looks good to me, @balisujohn ?

Reply:
I think it might be good to create a test that creates a local minari dataset using a dummy env then makes sure that when the dataset is converted, the observations etc... are in the right order in the resulting agileRL dataset. I also think it would be good to confirm we can train a policy via agileRL via a loaded Minari dataset(maybe a locally generated cartpole one). Otherwise, barring small stuff, looks good overall.

Reply:
Crucially, I should add that the integration seems to be using the Minari API appropriately.

Reply:
@balisujohn i was waiting on the agile unit test structure to finalize @nicku-a can i go ahead with writing a test for the minari integration separately??

Reply:
Yep go ahead and create it